### PR TITLE
remove the bos token from dpo outputs

### LINKED ISSUE #1616: DPO Trainer Incorrectly Inserts BoS Before Chosen and Rejected Prompts for Llama3
### Please check that this issue hasn't been reported before.

- [X] I searched previous [Bug Reports](https://github.com/OpenAccess-AI-Collective/axolotl/labels/bug) didn't find any similar reports.

### Expected Behavior

The BoS should only appear at the start of the prompt.

### Current behaviour

The BoS token is inserted at the start of the prompt and also at the start of the Chosen and Rejected prompts.

[2024-05-13 19:18:27,809] [INFO] [axolotl.check_rl_example_labels:91] [PID:718] [RANK:0] INPUT PROMPT: <|begin_of_text|>(128000) 

[2024-05-13 19:18:27,809] [INFO] [axolotl.check_rl_example_labels:92] [PID:718] [RANK:0] CHOSEN RESPONSE: <|begin_of_text|>(128000) 

[2024-05-13 19:18:27,809] [INFO] [axolotl.check_rl_example_labels:93] [PID:718] [RANK:0] REJECTED RESPONSE: <|begin_of_text|>(128000) 

### Steps to reproduce

Run a DPO tune using intel.chatml. Preprocess the dataset with --debug flag and you'll see that the BoS token is outputted.

### Config yaml

```yaml
rl: dpo
datasets:
  - ds_type: json
    data_files: 
      - combinedDPO.json
    split: train
    type: chatml.intel
```

### Possible solution

_No response_

### Which Operating Systems are you using?

- [X] Linux
- [ ] macOS
- [ ] Windows

### Python Version

Whatever version the latest docker uses.

### axolotl branch-commit

The latest commit that the docker is using.

### Acknowledgements

- [X] My issue title is concise, descriptive, and in title casing.
- [X] I have searched the existing issues to make sure this bug has not been reported yet.
- [X] I am using the latest version of axolotl.
- [X] I have provided enough information for the maintainers to reproduce and diagnose the issue.

### PR DESCRIPTION
**Pull request by winglian**
Fixes #1616

### CODE REVIEW

File: src/axolotl/core/trainer_builder.py [hunk 1]
```diff
@@ -963,9 +963,25 @@ def tokenize_row(
...
+                res["chosen_labels"] = res["chosen_labels"][1:]
+                res["chosen_attention_mask"] = res["chosen_attention_mask"][1:]
+            if res["rejected_input_ids"][0] == self.tokenizer.bos_token_id:
+                res["rejected_input_ids"] = res["rejected_input_ids"][1:]
+                res["rejected_labels"] = res["rejected_labels"][1:]
+                res["rejected_attention_mask"] = res["rejected_attention_mask"][1:]
+            if res["prompt_input_ids"][0] == self.tokenizer.bos_token_id:
+                res["prompt_input_ids"] = res["prompt_input_ids"][1:]
+                res["prompt_attention_mask"] = res["prompt_attention_mask"][1:]
+
```
Comment (NanoCode012) [inline src/axolotl/core/trainer_builder.py:1071]: From the linked Issue, we should only remove the BOS from the `chosen`/`rejected`, right? The `prompt` should not require it?

If it does require removing bos, does `res["prompt_labels"]` exist & do we need to remove it also?
